test/test_torch.py

```diff
@@ -15375,6 +15376,25 @@ def test_max(self, device, dtype):
     def test_min(self, device, dtype):
         self._test_minmax_helper(torch.min, np.amin, device, dtype)
 
+    @dtypes(torch.float)
+    def test_min_out_different_memory_format(self, device, dtype):
+        permutations = itertools.permutations([0, 1, 2, 3])
+
+        def rev_perm(perm):
+            r = {j: i for i, j in enumerate(perm)}
+            return tuple(r[i] for i, _ in enumerate(perm))
+
+        for perm1 in permutations:
+            for perm2 in permutations:
+                for perm3 in permutations:
+                    input_ = torch.randn(10, 10, 10, 1, dtype=dtype, device=device).expand(10, 10, 10, 10).permute(*perm1)
+                    expect1, expect2 = input_.min(dim=0, keepdim=True)
+                    out1 = expect1.permute(perm2).clone().contiguous().permute(rev_perm(perm2))
```

Comment:
It could be `empty_like(expect1.permute(perm2)).contiguous().permute(rev_perm(perm2))`
I need to do a permute and then permute it back so that the shape is unchanged, but the strides are permuted.

Reply:
I replaced it with `torch.empty_like(expect1.permute(perm2), memory_format=torch.contiguous_format)`